crates/uv-resolver/src/resolution/graph.rs

```diff
@@ -248,23 +252,30 @@ impl ResolutionGraph {
                         continue;
                     };
 
-                    let PubGrubPackageInner::Package {
-                        name: dependency_name,
-                        extra: dependency_extra,
-                        ..
-                    } = &**dependency_package
-                    else {
-                        continue;
-                    };
+                    match &**dependency_package {
+                        PubGrubPackageInner::Package {
+                            name: dependency_name,
+                            extra: dependency_extra,
+                            ..
+                        } => {
+                            let self_index = &inverse[&(self_name, self_extra)];
+                            let dependency_index = &inverse[&(dependency_name, dependency_extra)];
+                            petgraph.update_edge(*self_index, *dependency_index, ());
+                        }
 
-                    // For extras, we include a dependency between the extra and the base package.
-                    if self_name == dependency_name {
-                        continue;
-                    }
```

Comment:
How come this check isn't needed any more?

Reply:
I think this was made redundant in a prior change, when we introduced `PubGrubPackageInner::Inner`.

Reply:
I'll pull this into a separate change for sanity.